src/mmrelay/matrix/sync_bootstrap.py

```
@@ -0,0 +1,1199 @@
+        client (AsyncClient): Connected Matrix client used for alias resolution and state inspection.
+        sync_response (Any | None): Result of the initial sync; if it represents a sync error the function aborts.
+        config_data (dict[str, Any]): Configuration data used to evaluate E2EE settings and for display output.
+        matrix_rooms (Any): In-memory matrix_rooms mapping (list or dict); room aliases found here will be resolved in-place to canonical room IDs.
+        matrix_homeserver (str): Homeserver URL used for diagnostic logging when sync errors occur.
+        bot_user_id (str): The bot's Matrix user ID used to fetch and store the bot display name.
+        e2ee_enabled (bool): True when end-to-end encryption is intended for this client; client.e2ee_enabled is set to this value.
+
+    Raises:
+        MatrixSyncFailedDetailsError: If the provided sync_response indicates a SyncError; error type and a short, user-facing error message are included.
+    """
+    global bot_user_name
```

Ruff (0.15.9)

[warning] Using global for `bot_user_name` but no assignment is done

(PLW0602)